src/core/gui/Builder.cpp

```diff
@@ -0,0 +1,10 @@
+#include "Builder.h"
+
+#include "GladeSearchpath.h"  // for GladeSearchpath
+#include "filesystem.h"       // for path
+
+Builder::Builder(GladeSearchpath* gladeSearchPath, const std::string& uiFile):
+        builder(gtk_builder_new_from_file(gladeSearchPath->findFile("", uiFile).u8string().c_str()), xoj::util::adopt) {
+}
+
+Builder::~Builder() = default;
```

Comment:
While defaulted destructors and (in some cases) constructors allow a forward declaration of incomplete field types, it also prevents several optimizations and forces the programmer to use the rule of 5.

Therefore, the rule of thumb is, to remove all defaulted special members, unless they must be declared + defined. Try to define them in the header.
```suggestion
```